numpy/linalg/tests/test_linalg.py

```diff
@@ -845,6 +846,111 @@ def test_empty(self):
         assert_equal(norm(array([], dtype=self.dt)), 0.0)
         assert_equal(norm(atleast_2d(array([], dtype=self.dt))), 0.0)
 
+    def test_vector_types(self):
+        a = np.array([1, 0, 1])
+
+        exact_types = [
```

Comment:
You could more precise, perhaps, and check that integer types convert to doubles, and floats to 'f' kind.